### PR TITLE
[flang][Driver] Support -fdiagnostics-color

### PR DESCRIPTION
**Pull request by tarunprabhu**
Add support for -fdiagnostics-color and -fdiagnostics-color=. Add documentation for -fdiagnostics-color= which should also be visible in clang.

Partially addresses requests in #89888

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Flang.cpp [hunk 1]
```diff
@@ -732,6 +732,12 @@ void Flang::ConstructJob(Compilation &C, const JobAction &JA,
   // to avoid warn_drv_unused_argument.
   Args.getLastArg(options::OPT_fcolor_diagnostics,
                   options::OPT_fno_color_diagnostics);
+  if (const Arg *A = Args.getLastArg(options::OPT_fdiagnostics_color_EQ)) {
+    StringRef Value(A->getValue());
+    if (Value != "always" && Value != "never" && Value != "auto")
+      D.Diag(diag::err_drv_invalid_argument_to_option)
+          << Value << A->getOption().getName();
+  }
```
Comment (kiranchandramohan) [inline clang/lib/Driver/ToolChains/Flang.cpp:740]: Should this code be shared with Clang?

**Review by tblah — COMMENTED** [review]
Is `-fdiagnostics-color=auto`  too host-environment dependent to test?

**Comment by tarunprabhu** [comment]
> Is `-fdiagnostics-color=auto` too host-environment dependent to test?

I am not sure how I could reliably test it across platforms. One way to do it on Unix could be 

```
$ flang -### -diagnostics-color=auto %s | grep -Eqv '"-fc1".+"-fcolor-diagnostics"'
```

The pipe to grep would turn colors off. 

On the other hand, I don't know if  `REQUIRES: shell` ensures that the shell always has color support. If it does not guarantee that, the "normal" approach - similar to the way `-fdiagnostics-color=always` is tested - might not work. I don't have access to any non-fairly-modern-Linux'es on which I could test this either.

**Review by MaskRay — APPROVED** [review]

**Comment by tarunprabhu** [comment]
Thanks for the review, @MaskRay. @tblah, @kiranchandramohan, does this look ok to you?

**Review by kiranchandramohan — APPROVED** [review]
LG. Thanks Tarun.

**Pull request merged**